trajectory_executor/com_serial.py

```diff
@@ -176,12 +180,22 @@ def sensor_data_request_callback(self, request: SensorDataRequest, response: Sen
             break
           else:
             # bad packet, retry request
-            return self.sensor_data_request_callback(request, response)
+            # return self.sensor_data_request_callback(request, response)
+            break
+
+      else:
+        if read_corrupted:
+          # self.get_logger().info("Read corrupted. Aborting")
+          break
+        else:
+          read_corrupted = True # gives a second chance in case we just missed a beat
```

Comment:
I am a little unclear on how this read_corrupt logic is working. Can you explain a little bit?

Reply:
So reading the serial port in this node is encapsulated in an "infinite" loop. Previously the only exit condition was when a full packet was received. What started happening when I was testing was that the serial read would get stuck at one point in the loop and completely block all serial comms even though there was no data incoming. This `read_corrupted` flag serves as a way to abort the loop in the event that happens. If we're in the loop and there's nothing on the serial port, it goes to this `else` statement where the `read_corrupted` flag is first set but doesn't break out of the loop yet. This offers the opportunity for data to get on the serial port in case we entered the loop too quickly somehow. If there's still no data on the port, then with `read_corrupted` set to `True`, it aborts the loop